api/main.py

```diff
@@ -16,41 +19,56 @@ def student_id_positive(cls, value):
             raise ValueError(f"Expected positive price, received {value}")
         return value
 
+    @validator("student_id")
+    def student_id_unique(cls, value):
+        if value in [student["student_id"] for student in students]:
+            raise ValueError(f"ID is already in use")
+        return value
+
 @app.get("/")
 def read_root():
     return {"Hello": "World"}
 
+
 @app.get("/items/{item_id}")
-def read_item(item_id: int, q: Union[str, None] = None):
+def read_item(item_id: int, q: str):
     name = q.upper()
     return {"item_id": item_id, "q": name}
 
+
 @app.get("/name/{q}")
 def read_name(q: str):
-    return (q.upper())
+    return q.upper()
+
 
 @app.get("/students")
 def read_students():
     return students
 
+
 @app.post("/students")
 def create_student(student: Student):
     return student
 
+
 @app.put("/students/{student_id}")
 def save_student(student: Student):
     return student
 
+
 @app.patch("/students")
 def edit_student(student: Student):
-    students[student.student_id-1]["name"] = student.name
+    for person in students:
+        if person["student_id"] == student.student_id:
+            person["name"] = student.name
     return students
 
+
 @app.delete("/students/{student_id}")
 def delete_student(student_id: int):
     for i in range(len(students)):
         if students[i]["student_id"] == student_id:
             del students[i]
             break
     print(students)

```

Comment:
prints are only for development, we avoid them in production --> instead we use loggers
I would delete it :)

Reply:
I'll delete it! :)